### PR TITLE
chore: Datadog PoC [OTE-702] [2/n]

### PR DESCRIPTION
**Pull request by yogurtandjam**
Adds initial events described by the [tech spec](https://www.notion.so/dydx/Web-Datadog-PoC-17ef48f266ae46a4951188015c404b5b)

- DepositSubmit
- WithdrawalSubmit
- DepositRouteErrors
- WithdrawalRouteErrors
- withdrawToNobleIBC
- cctpWithdraw
- cctpMultiMsgWithdraw
- sendNobleIBC

Also adds ddInfo and ddError methods to abacus logger implementation so we can pick them up in abacus and use them at will.

Finally, this PR adds `dd.error` to the `log` method, so any telemetry logs will automatically be piped into datadog

### CODE REVIEW

File: src/lib/abacus/logger.ts [hunk 1]
```diff
@@ -19,6 +21,20 @@ class AbacusLogger implements Omit<AbacusLoggingProtocol, '__doNotUseOrImplement
       console.error(`${tag}: ${message}`);
     }
   }
+
+  ddInfo(message: string, context: object) {
```
Comment (yogurtandjam) [inline src/lib/abacus/logger.ts:25]: @dydxprotocol/frontend this is how im planning on giving abacus the power to log info and error msgs to datadog.
it's a little hacky but until we're ready to merge their logs i figure this is good enough.

File: src/lib/analytics/amplitude.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const identify = (property: AnalyticsUserProperty) => {
     detail: { property: propertyTypeToLog, propertyValue: property.payload },
   });
   dd.setContextProperty(propertyTypeToLog, property.payload);
-  dd.log(`set context item: ${propertyTypeToLog}`, dd.getContext());
+  dd.info(`set context item: ${propertyTypeToLog}`, dd.getContext());
```
Comment (rosepuppy) [inline src/lib/analytics/amplitude.ts:25]: Does it make sense to just add `dd.info` to `track`? or was it decided that is too much events

Comment (yogurtandjam) [reply src/lib/analytics/amplitude.ts:25]: this was discussed [in slack](https://dydx-team.slack.com/archives/C042V24U87Q/p1724109154414829?thread_ts=1724101364.841789&cid=C042V24U87Q) but we decided to be very conservative about the # of events we send and track the volume first

File: src/lib/abacus/logger.ts [hunk 3]
```diff
@@ -11,13 +13,21 @@ class AbacusLogger implements Omit<AbacusLoggingProtocol, '__doNotUseOrImplement
     }
   }
 
-  e(tag: string, message: string) {
+  e(tag: string, message: string, context: object, error: Error) {
```
Comment (yogurtandjam) [inline src/lib/abacus/logger.ts:16]: this requires us to update abacus and bump the version

File: src/lib/abacus/logger.ts [hunk 4]
```diff
@@ -11,21 +13,26 @@ class AbacusLogger implements Omit<AbacusLoggingProtocol, '__doNotUseOrImplement
...
+  ddInfo(tag: string, message: string, context: string) {
+    try {
+      const parsedContext = JSON.parse(context);
+      if (import.meta.env.VITE_DATADOG_LOG_LEVEL === 'debug') {
+        console.log(`${tag} dd info: ${message}`, parsedContext);
+      }
+      dd.info(message, parsedContext);
+      // catch in case parsing context fails for some reason
+    } catch (err) {
+      dd.error('Error sending dd info', undefined, err);
```
Comment (tinaszheng) [inline src/lib/abacus/logger.ts:35]: maybe send the context string in this case?

Comment (yogurtandjam) [reply src/lib/abacus/logger.ts:35]: ah good point, forgot the error won't have that data. i like it

**Review by tinaszheng — APPROVED** [review]

**Pull request merged**